engines/prover.h

```diff
@@ -25,6 +25,17 @@
 #include "smt-switch/smt.h"
 
 namespace pono {
+
+/** enum for communicating result of a refinement step
+ *  only used for algorithms that use abstraction refinement
+ */
+enum RefineResult
```

Comment:
These are just supposed to be used by CEGAR loops to communicate how refinement went. In IC3Base, `refine` will return `REFINE_NONE` by default -- meaning no refinement is necessary, so a counterexample is always real unless this is overridden.